packages/common/src/components/Console/index.tsx

```diff
@@ -132,74 +128,145 @@ class Console extends React.Component<IPrivateProps, IState> {
-                      lineHeight: '1.2rem',
-                    }}
-                  />
+          {items.length === 0 ? (
+            <NoLogsPlaceholderContainer>
+              <NoLogsPlaceholder>
+                There are no logs to display. Use{' '}
+                <pre
+                  style={{
+                    fontFamily: 'Consolas, monaco, monospace',
+                    fontWeight: 'bold',
+                    display: 'inline',
+                  }}
+                >
+                  console.log()
+                </pre>{' '}
+                to display logs here.
+              </NoLogsPlaceholder>
+            </NoLogsPlaceholderContainer>
+          ) : (
+            <LogsArea>
+              {items.map(({ backgroundColor, color, key, icon, message }) =>
+                typeof message === 'object' ? (
+                  <ObjectInspectorLogEntry
+                    key={key}
+                    backgroundColor={backgroundColor}
+                    style={{ backgroundColor, color }}
+                  >
+                    {icon ? (
+                      <Icon
+                        className="ms-font-m"
+                        iconName={icon.name}
+                        style={{
+                          fontSize: '1.2rem',
+                          color: icon.color,
+                          lineHeight: '1.2rem',
+                        }}
+                      />
+                    ) : (
+                      <div style={{ width: '1.2rem', height: '1.2rem' }} />
+                    )}
+                    {getObjectInspectorForData(message)}
+                  </ObjectInspectorLogEntry>
                 ) : (
-                  <div style={{ width: '1.2rem', height: '1.2rem' }} />
-                )}
-                <LogText>{message}</LogText>
-              </Log>
-            ))}
-            <div ref={this.lastLog} />
-          </LogsArea>
+                  <LogEntry key={key} style={{ backgroundColor, color }}>
+                    {icon ? (
+                      <Icon
+                        className="ms-font-m"
+                        iconName={icon.name}
+                        style={{
+                          fontSize: '1.2rem',
+                          color: icon.color,
+                          lineHeight: '1.2rem',
+                        }}
+                      />
+                    ) : (
+                      <div style={{ width: '1.2rem', height: '1.2rem' }} />
+                    )}
+                    <LogText>{stringifyPlusPlusOrErrorMessage(message)}</LogText>
+                  </LogEntry>
+                ),
+              )}
+
+              <div ref={this.lastLog} />
+            </LogsArea>
+          )}
         </HeaderFooterLayout>
       </Wrapper>
     );
   }
 }
 
 export default withTheme(Console);
+
+///////////////////////////////////////
+
+function getObjectInspectorForData(obj: any): React.ReactElement<any> {
```

Comment:
This should be its own component

Reply:
That's some very solid advice :-).  I guess next time I write a "///////////////////////////" followed by a thing that returns a React component, maybe I should think about whether it just _should_ be a React component... :-)  Will fix.